Tools/autotest/arducopter.py

```diff
@@ -7664,39 +7664,309 @@ def AP_Avoidance(self):
+            0, # timestamp
+            1, # target system_id
+            1, # target component id
+            mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT_INT, # relative altitude frame
+            target_typemask | MAV_POS_TARGET_TYPE_MASK.LAST_BYTE, # target typemask as pos and vel only
+            int(location.lat * 1e7), # lat
+            int(location.lng * 1e7), # lon
+            location.alt, # alt
+            0, # vx
+            0, # vy
+            0, # vz
+            0, # afx
+            0, # afy
+            0, # afz
+            0, # yaw
+            0) # yawrate
+
+        # wait vehicle to move away from home
+        self.delay_sim_time(seconds_to_wait=10)
+
+        # pause guided flight
+        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
+                     0, # param1
+                     0, # param2
+                     0, # param3
+                     0, # param4
+                     0, # param5
+                     0, # param6
+                     0) # param7
+
+        # wait vehicle to pause
+        self.wait_groundspeed(0, 1, minimum_duration=5)
+
+        # continue guided flight
+        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
+                     1, # param1
+                     0, # param2
+                     0, # param3
+                     0, # param4
+                     0, # param5
+                     0, # param6
+                     0) # param7
+
+        # wait vehicle to reach destination
+        self.wait_location(loc=location, timeout=120)
+
+        # end pause/continue subtest with destination and velocity
+        self.end_subtest("Ended test for Pause/Continue in GUIDED mode with DESTINATION, VELOCITY!")
+
+        # start pause/continue subtest with destination, velocity and acceleration
+        self.start_subtest("Started test for Pause/Continue in GUIDED mode with DESTINATION, VELOCITY, ACCELERATION!")
+
+        # point to north
+        self.guided_achieve_heading(0)
+
+        # location to test
+        location = self.offset_location_ne(self.mav.location(), 300, 0)
+
+        # target typemask as pos, vel and acc only
+        target_typemask = (MAV_POS_TARGET_TYPE_MASK.YAW_IGNORE |
+                           MAV_POS_TARGET_TYPE_MASK.YAW_RATE_IGNORE)
+
+        # move vehicle on x direction
+        self.mav.mav.set_position_target_local_ned_send(
+            0, # system time in milliseconds
+            1, # target system
+            1, # target component
+            mavutil.mavlink.MAV_FRAME_BODY_NED, # coordinate frame MAV_FRAME_BODY_NED
+            target_typemask | MAV_POS_TARGET_TYPE_MASK.LAST_BYTE, # target typemask as pos, vel and acc only
+            300, # position x
+            0, # position y
+            0, # position z
+            0, # velocity x
+            0, # velocity y
+            0, # velocity z
+            0, # accel x
+            0, # accel y
+            0, # accel z
```

Comment:
You set the mask but our targets at the end of reaching the position are all 0's...
I dont think it is required to get this in but we should some how test non-zero velocity targets and accelerations targets as well. 

IE ensure when pausing under velocity control or acceleration control that the pausing works as you expect.